content/zh/docs/tasks/run-application/horizontal-pod-autoscale.md

```diff
@@ -0,0 +1,554 @@
+
+<!--
+If there were any missing metrics, we recompute the average more
+conservatively, assuming those pods were consuming 100% of the desired
+value in case of a scale down, and 0% in case of a scale up.  This dampens
+the magnitude of any potential scale.
+-->
+如果有任何 pod 的指标缺失，我们会更保守地重新计算平均值，
+在需要缩小时假设这些 pod 消耗了目标值的 100%，
+在需要放大时假设这些 pod 消耗了0%目标值。
+这可以在一定程度上抑制伸缩的幅度。
+
+<!--
+Furthermore, if any not-yet-ready pods were present, and we would have
+scaled up without factoring in missing metrics or not-yet-ready pods, we
+conservatively assume the not-yet-ready pods are consuming 0% of the
+desired metric, further dampening the magnitude of a scale up.
+-->
+此外，如果存在任何尚未就绪的pod，我们可以在不考虑遗漏指标或尚未就绪的pods的情况下进行伸缩，
+我们保守地假设尚未就绪的pods消耗了试题指标的0%，从而进一步降低了伸缩的幅度。
+
+<!--
+After factoring in the not-yet-ready pods and missing metrics, we
+recalculate the usage ratio.  If the new ratio reverses the scale
+direction, or is within the tolerance, we skip scaling.  Otherwise, we use
+the new ratio to scale.
+-->
+在缩放方向（缩小或放大）确定后，我们会把未就绪的 pod 和缺少指标的 pod 考虑进来再次计算使用率。
+如果新的比率与缩放方向相反，或者在容忍范围内，则跳过缩放。
+否则，我们使用新的缩放比例。
+
+<!--
+Note that the *original* value for the average utilization is reported
+back via the HorizontalPodAutoscaler status, without factoring in the
+not-yet-ready pods or missing metrics, even when the new usage ratio is
+used.
+-->
+注意，平均利用率的*原始*值会通过 HorizontalPodAutoscaler 的状态体现（
+即使使用了新的使用率，也不考虑未就绪 pod 和 缺少指标的 pod)。
+
+<!--
+If multiple metrics are specified in a HorizontalPodAutoscaler, this
+calculation is done for each metric, and then the largest of the desired
+replica counts is chosen.  If any of those metrics cannot be converted
+into a desired replica count (e.g. due to an error fetching the metrics
+from the metrics APIs), scaling is skipped.
+-->
+如果创建 HorizontalPodAutoscaler 时指定了多个指标，
+那么会按照每个指标分别计算缩放副本数，取最大的进行缩放。
+如果任何一个指标无法顺利的计算出缩放副本数（比如，通过 API 获取指标时出错），
+那么本次缩放会被跳过。
+
+<!--
+Finally, just before HPA scales the target, the scale recommendation is recorded.  The
+controller considers all recommendations within a configurable window choosing the
+highest recommendation from within that window. 
+This value can be configured using the `--horizontal-pod-autoscaler-downscale-stabilization` flag, which defaults to 5 minutes.
+This means that scaledowns will occur gradually, smoothing out the impact of rapidly
+fluctuating metric values.
+-->
+最后，在 HPA 控制器执行缩放操作之前，会记录缩放建议信息（scale recommendation）。
+控制器会在操作时间窗口中考虑所有的建议信息，并从中选择得分最高的建议。
+这个值可通过 kube-controller-manager 服务的启动参数 `--horizontal-pod-autoscaler-downscale-stabilization` 进行配置，
+默认值为 5min。
+这个配置可以让系统更为平滑地进行缩容操作，从而消除短时间内指标值快速波动产生的影响。
+
+<!--
+## API Object
+-->
+## API 对象
+
+<!--
+The Horizontal Pod Autoscaler is an API resource in the Kubernetes `autoscaling` API group.
+The current stable version, which only includes support for CPU autoscaling,
+can be found in the `autoscaling/v1` API version.
+-->
+Horizontal Pod Autoscaler 是 Kubernetes `autoscaling` API 组的资源。
```

Comment:
```suggestion
HorizontalPodAutoscaler 是 Kubernetes `autoscaling` API 组的资源。
```
(“Horizontal Pod Autoscaler”, with spaces, works in English but not in other locales).

Reply:
Why can not works in other locales?  example ?

Reply:
The resource is called HorizontalPodAutoscaler everywhere, but the three English words “Horizontal Pod Autoscaler” don't represent three resources called Horizontal, Pod, and Autoscaler.
For that reason, remove the spaces to yield the correct resource name: HorizontalPodAutoscaler

As an alternative, you might use something like:
```markdown
Pod 水平自动伸缩是 Kubernetes `autoscaling` API 组的资源。
```
(sorry about my grammar there!)

Reply:
Accepted.
Thanks @sftim :)